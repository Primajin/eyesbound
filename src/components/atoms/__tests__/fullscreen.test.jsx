import React from 'react';
import {act, create} from 'react-test-renderer';

import Fullscreen from '../fullscreen.jsx';

describe('Fullscreen', () => {
	it('renders correctly without props', () => {
		const component = create(<Fullscreen/>);
		expect(component).toMatchSnapshot();
	});

	it('renders correctly with props', () => {
		const callback = jest.fn();
		const selector = '';
		const component = create(<Fullscreen callback={callback} selector={selector}/>);
		expect(component).toMatchSnapshot();
	});

	it('calls back the callback', async () => {
		/* global document */
		document.addEventListener = jest.fn((_, callback) => {
			callback();
		});
		const callback = jest.fn();
		const selector = '';
		const component = create(<Fullscreen callback={callback} selector={selector}/>);
		await act(() => {
			component.root.findByType('button').props.onClick();
			globalThis.document.dispatchEvent(new Event('fullscreenchange'));
		});
		expect(callback).toHaveBeenCalledTimes(1);
	});
});
