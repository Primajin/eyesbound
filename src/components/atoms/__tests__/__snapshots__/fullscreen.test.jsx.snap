// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fullscreen renders correctly with props 1`] = `
<button
  aria-label="Toggle fullscreen"
  className=""
  css={
    Object {
      "map": undefined,
      "name": "12oymiu",
      "next": undefined,
      "styles": "
	
	align-items: center;
	background-color: var(--foreground);
	color: var(--background);
	cursor: pointer;
	display: flex;
	height: 50px;
	justify-content: center;
	opacity: 1;
	padding: 0;
	position: fixed;
	right: 50px;
	top: 0;
	transition: background-color .333s, color .333s, opacity .333s;
	width: 50px;
	z-index: 8;

	&:hover {
		background: var(--background);
		color: var(--foreground);
	}

	&.fullScreen {
		opacity: 0.65;
	}

	svg {
		fill: currentColor;
		height: 40px;
		width: 40px;
	}
;;
	right: 0;

	svg {
		height: 50px;
		width: 50px;
	}
",
      "toString": [Function],
    }
  }
  onClick={[Function]}
  tabIndex="12"
  type="button"
>
  <svg
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
    />
  </svg>
</button>
`;

exports[`Fullscreen renders correctly without props 1`] = `
<button
  aria-label="Toggle fullscreen"
  className=""
  css={
    Object {
      "map": undefined,
      "name": "12oymiu",
      "next": undefined,
      "styles": "
	
	align-items: center;
	background-color: var(--foreground);
	color: var(--background);
	cursor: pointer;
	display: flex;
	height: 50px;
	justify-content: center;
	opacity: 1;
	padding: 0;
	position: fixed;
	right: 50px;
	top: 0;
	transition: background-color .333s, color .333s, opacity .333s;
	width: 50px;
	z-index: 8;

	&:hover {
		background: var(--background);
		color: var(--foreground);
	}

	&.fullScreen {
		opacity: 0.65;
	}

	svg {
		fill: currentColor;
		height: 40px;
		width: 40px;
	}
;;
	right: 0;

	svg {
		height: 50px;
		width: 50px;
	}
",
      "toString": [Function],
    }
  }
  onClick={[Function]}
  tabIndex="12"
  type="button"
>
  <svg
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
    />
  </svg>
</button>
`;
